core/src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala

```diff
@@ -356,6 +358,7 @@ abstract class ConvertToDeltaCommandBase(
         spark.sessionState.catalog
       )
     }
+    if (collectStats) { StatisticsCollection.recompute(spark, txn.deltaLog) }
```

Comment:
hm. so this would be done in the **next** commit? what if there's a commit in-between these two?

Reply:
Good question. 

So files that were added after the first commit of the convert, should already have statistics in case the default configuration wasn't changed (which makes me think that I should also check if the configuration is set to collect statistics before actually collecting them. This raises another question - what if collectStats is true but config is set to false? Should we raise an exception here or simply avoid collecting stats during convert?)

So the files on first commit does not have stats, files on second commit (the one between the convert and compute-stats) have statistics by default, now `StatisticsCollection.recompute` should take all the files from the log and collect their statistics (I think that the files from the second commit will be re-computed).

Eventually all the files should have statistics after the third commit.

I tested it locally with

1. writing parquet files
2. convert to delta without statistics
3. write delta files
4. recompute statistics
5. all files had statistics

I can add this as an additional unit test.

Are you asking because maybe the `txn.deltaLog` is not updated with the latest files from commit 2? I admit I'm not yet familiar with the code base so I'm not sure if that is possible and how, but maybe `StatisticsCollection.recompute(spark, txn.deltaLog.update().deltaLog)` should make sure no files will be missing?

Reply:
Sorry for the delay in response.

> what if collectStats is true but config is set to false? Should we raise an exception here or simply avoid collecting stats during convert?

Let's throw an exception. This is a weird state, we don't want to confuse the user.

> Are you asking because maybe the txn.deltaLog is not updated with the latest files from commit 2?

I'm asking because of the edge case of 
```
commit 0 -> convert parquet to delta
commit 1 -> remove a file
commit 2 -> compute stats finally finishes. COLLISION.
```

> Are you asking about the possibility of a conflict on the next commit (the stats commit) which could eventually throw an exception? So in that case we'll have a converted delta table but without statistics even though collectStats was set to true? If so -> Should we raise another, more specific exception do describe the situation? Or maybe call recompute again?

No don't throw an exception. And don't call recompute again. I think this case is very, very rare.

But ...... I do want to think about if we can do this in only 1 commit instead of 2.

Reply:
I've figured out a simple way to write the stats in 1 commit instead of 2.

Let's add this to the `ConvertToDeltaCommand` object:
```scala
def computeStats(
      deltaLog: DeltaLog,
      snapshot: Snapshot,
      addFiles: Seq[AddFile]): Iterator[AddFile] = {
    val filesWithStats = deltaLog.createDataFrame(snapshot, addFiles)
      .groupBy(input_file_name()).agg(to_json(snapshot.statsCollector))

    val pathToAddFileMap = generateCandidateFileMap(deltaLog.dataPath, addFiles)
    filesWithStats.collect().iterator.map { row =>
      val addFile = getTouchedFile(deltaLog.dataPath, row.getString(0), pathToAddFileMap)
      addFile.copy(stats = row.getString(1))
    }
  }
```

and then we can call it from inside of `ConvertToDeltaCommandBase::createDeltaActions`.

i.e. if if we should collect stats, return the method above, else inside of `createDeltaActions` we just return the current existing `adds.toIterator`.

make sense?